src/main.rs

```diff
@@ -269,14 +273,15 @@ impl App {
         let (_, content) = self.files.get(&params.text_document.uri)?;
 
         let mut completions = Vec::new();
-        for var in scope.keys() {
+        for (var, (datatype, _)) in scope {
             if var.starts_with(&name.as_str()) {
                 completions.push(CompletionItem {
                     label: var.clone(),
                     text_edit: Some(TextEdit {
                         range: utils::range(content, name.node().text_range()),
                         new_text: var.clone(),
                     }),
+                    detail: Some(datatype),
```

Comment:
The `datatype` variable should probably be an enum implementing Display instead, to save memory usage I guess.

Reply:
Updated.